lib/ansible/modules/windows/win_wait_for_process.ps1

```diff
@@ -64,99 +64,108 @@ Function Get-ProcessMatchesFilter {
         [int]
         $ProcessId
     )
-    
-    $CIMProcesses = Get-CimInstance Win32_Process
-    foreach ($CIMProcess in $CIMProcesses)
-    {
-        $include = $true
-        if(-not [String]::IsNullOrEmpty($ProcessNamePattern))
-        {
-            #if a process name was specified in the filter, validate that here.
-            $include = $include -and ($CIMProcess.ProcessName -match $ProcessNamePattern)
-        }
-        if($ProcessNameExact -is [Array] -or (-not [String]::IsNullOrEmpty($ProcessNameExact)))
-        {
-            #if a process name was specified in the filter, validate that here.
-            if ($ProcessNameExact -is [Array] )
-            {
-                $include = $include -and ($ProcessNameExact -contains $CIMProcess.ProcessName)
-            }   
-            else {
-                $include = $include -and ($ProcessNameExact -eq $CIMProcess.ProcessName)
+
+    $WMIProcesses = Get-Process -IncludeUserName
```

Comment:
To keep compatibility with PSv3 you should optionally include the `-IncludeUserName` parameter depends on if `$Owner` is set. This way you can say in the docs, the `owner` option requires PSv4+ but the module will still work with other ones if you are on that version.

```
$parameters = @{}
if ($Owner) {
    $parameter.IncludeUserName = $true
}
$WMIProcesses = Get-Process @parameters
```

Also WMIProcesses isn't a good name, this get's all processes and not WMI ones.

Reply:
It used to be called $CIMProcesses, that's why I named it $WMIProcesses. But yeah, could as well be $Processes.